### DISCUSSION TITLE
Check for availability of Processor/Language in a Java Loader

### DISCUSSION THREAD

**Discussion by hippietrail** [NationalSecurityAgency/ghidra#6968] [Q&A] [answered]
I'm working on some Loaders which will use a Processor Language extension available on GitHub if it's installed. I don't want it to throw an exception and give up if it's not available so I try to check for it first. This is the best way I've found but my gut feeling is it's probably not the cleanest way to do it but I can't find any other Loader doing anything similar:
```java
	LanguageCompilerSpecPair lcsp = new LanguageCompilerSpecPair("9900:BE:16:default", "default");

	LanguageID languageID = lcsp.getLanguageID();

	try {
		Language language = getLanguageService().getLanguage(languageID);
	} catch (Exception e) {
		Msg.warn(this, "TMS 9900 support not found. Find the extension on GitHub.");
	}

	loadSpecs.add(new LoadSpec(this, 0, lcsp, true));
```
Is there a better way to do this?

**Comment by dev747368** [comment, marked as answer]
My $0.02 is that its not that bad.

Calling ```getLanguageDescription()``` instead of ```getLanguage()``` might be more lightweight if you are not going to use the result immediately.
